experimental/GModule/GModule.jl

```diff
@@ -207,8 +212,71 @@ where `phi` is an embedding of `S` into `R`.
 
 (case that `R` is a number field and `S` is the ring of integers in `R`)
 """
-function invariant_lattice_classes(M::GModule, phi::Map)
-  error("not yet ...")
+function invariant_lattice_classes(M::GModule{<:Oscar.GAPGroup, <:AbstractAlgebra.FPModule{QQFieldElem}}, phi::Map{ZZRing, QQField})
+  MZ = gmodule(ZZ, M)
+  return invariant_lattice_classes(MZ)
+end
+
+function Oscar.pseudo_inv(h::Generic.ModuleHomomorphism)
+  return MapFromFunc(codomain(h), domain(h), x->preimage(h, x))
+end
+
+function _hom(f::Map{<:AbstractAlgebra.FPModule{T}, <:AbstractAlgebra.FPModule{T}}) where T
+  @assert base_ring(domain(f)) == base_ring(codomain(f))
+  return hom(domain(f), codomain(f), f.(gens(domain(f))))
+end
+
+function invariant_lattice_classes(M::GModule{<:Oscar.GAPGroup, <:AbstractAlgebra.FPModule{ZZRingElem}})
+  res = Any[(M, sub(M.M, gens(M.M))[2])]
+  sres = 1
+  new = true
+  lp = keys(factor(order(M.G)).fac)
+  while new
+    new  = false
+    lres = length(res)
+    for X = res[sres:end]
+      for p = lp
```

Comment:
I really don't like the `=` in iteration, would you mind changing it to
```suggestion
    for X in res[sres:end]
      for p in lp
```